src/main/java/com/google/sps/servlets/GetBlurAreasServlet.java

```diff
@@ -0,0 +1,186 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.FaceAnnotation;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.Vertex;
+import com.google.gson.Gson;
+import com.google.protobuf.ByteString;
+import java.awt.Point;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-blur-areas")
+public class GetBlurAreasServlet extends HttpServlet {
+
+  /**
+   * This method handles the POST requests to "/get-blur-areas". Receives a BlobKey parameter which
+   * represents the Blobstore location of an image. Responds with a JSON ArrayList of polygons. A
+   * polygon is represented by a List of points. A point contains two properties, its coordinates: x
+   * and y.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the BlobKey that points to the image uploaded by the user.
+    BlobKey blobKey = getBlobKey(request, "image");
+
+    // User didn't upload a file, so render an error message.
+    if (blobKey == null) {
+      response.getWriter().println("Please upload an image file.");
+      return;
+    }
+
+    // Get the image the user uploaded as bytes.
+    byte[] imageBytes = getBlobBytes(blobKey);
+
+    ArrayList<List<Point>> blurAreas = getBlurAreas(imageBytes);
+
+    // Delete the image from the blobstore.
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    blobstoreService.delete(blobKey);
+
+    // Convert the polygons to JSON.
+    Gson gson = new Gson();
+    String jsonResponse = gson.toJson(blurAreas);
+
+    // Send the JSON back as the response.
+    response.setContentType("application/json");
+    response.getWriter().println(jsonResponse);
+  }
+
+  /**
+   * Uses the Google Cloud Vision API to find faces in the image represented by the binary data
+   * stored in @param imageBytes
+   *
+   * @return an ArrayList of bounding polygons representing faces. Polygons are represented by a
+   *     list of points.
```

Comment:
Maybe you could say that the method returns an array of rectangles, not any polygons. It is essential for blurring method.